src/vs/workbench/contrib/chat/browser/chatContentParts/toolInvocationParts/chatTerminalToolProgressPart.ts

```diff
@@ -63,6 +72,40 @@ export class ChatTerminalToolProgressPart extends BaseChatToolInvocationSubPart
 		));
 		this._register(titlePart.onDidChangeHeight(() => this._onDidChangeHeight.fire()));
 
+		// Only show focus action when output location is set to 'none'
+		if (configurationService.getValue<string>('chat.tools.terminal.outputLocation') === 'none') {
+			let terminalInstance = getTerminalInstanceByToolSessionId(terminalData.terminalToolSessionId);
+			if (elements.actionBar) {
+				const actionBar = new ActionBar(elements.actionBar, {});
+				this._register(actionBar);
+				const focusAction = new Action(
+					TerminalCommandId.FocusInstance,
+					localize('runInTerminalTool.showTerminalButtonLabel', "Focus Terminal"),
+					ThemeIcon.asClassName(Codicon.linkExternal),
+					!!terminalInstance,
+					async () => {
+						if (terminalInstance) {
+							await commandService.executeCommand(TerminalCommandId.FocusInstance, terminalInstance);
+						}
+					}
+				);
+				focusAction.tooltip = localize('runInTerminalTool.focusTerminal', "Focus Terminal");
+				actionBar.push([focusAction], { icon: true, label: false });
+
+				if (!terminalInstance && terminalData.terminalToolSessionId) {
+					const listener = onDidRegisterTerminalInstanceForToolSession(e => {
+						if (e.terminalToolSessionId === terminalData.terminalToolSessionId) {
+							terminalInstance = getTerminalInstanceByToolSessionId(e.terminalToolSessionId);
+							if (terminalInstance) {
+								focusAction.enabled = true;
```

Comment:
Can we `this._store.delete(listener)` here to avoid holding onto that listener when we don't need it anymore?